driver-core/src/test/unit/com/mongodb/client/model/geojson/codecs/GeometryCollectionCodecSpecification.groovy

```diff
@@ -33,37 +41,146 @@ class GeometryCollectionCodecSpecification extends Specification {
     def writer = new BsonDocumentWriter(new BsonDocument())
     def context = EncoderContext.builder().build()
 
+    def 'should accept empty geometries'() {
+        def geometryCollection = new GeometryCollection([])
+
+        when:
+        codec.encode(writer, geometryCollection, context)
+
+        then:
+        writer.document == parse('{ type: "GeometryCollection", geometries: []}')
+
+        when:
+        def decodedGeometryCollection = codec.decode(new BsonDocumentReader(writer.document), DecoderContext.builder().build())
+
+        then:
+        geometryCollection == decodedGeometryCollection
+    }
+
     def 'should encode'() {
```

Comment:
Missed this the first time... Consider renaming all the "should encode" methods to "should round trip" or "should encode and decode"